src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SearchParameterQueryGenerator.cs

```diff
@@ -258,7 +258,22 @@ protected static SearchParameterQueryGeneratorContext VisitMissingFieldImpl(Miss
                 throw new InvalidOperationException($"Unexpected missing field {expression.FieldName}");
             }
 
-            AppendColumnName(context, column, expression).Append(" IS NULL");
+            switch (expression.FieldName)
+            {
+                case FieldName.TokenSystem:
+                case FieldName.QuantityCode:
+                    context.StringBuilder.Append(" ( ");
+                    AppendColumnName(context, column, expression).Append(" = 0");
```

Comment:
>0 [](http://example.com/codeflow?start=77&length=1)

nit:
I would suggest to use `NullSystemId` constant here.
I would also suggest to collapse `NullSystemId` and `NullQuantityCodeId` into one variable.

Reply:
Good point I will update this.